### PR TITLE
chore/remove-controller-errors

### LINKED ISSUE #123: Code smell: no error returned in any controller function
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

During https://github.com/caas-team/sparrow/issues/37 I noticed that the ChecksController doesn't ever return an error in any of its functions, as they are all throttled down the error channel.

When the err is always nil, then it should not be part of the function's signature :)

### Expected Behavior

No error as a return parameter.

### Steps To Reproduce

_No response_

### Relevant logs and/or screenshots, environment information, etc.

_No response_

### Who can address the issue?

Everyone

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by puffitos**
## Motivation

Closes #123

## Changes

Removed the errors from the controller functions, as they aren't/ produced at all.

For additional information look at the commits.

## Tests done

- [x] Unit tests succeeded

## TODO

- [x] I've assigned this PR to myself
- [x] I've labeled this PR correctly

### CODE REVIEW

File: pkg/sparrow/controller.go [hunk 1]
```diff
@@ -124,15 +123,15 @@ func (cc *ChecksController) Reconcile(ctx context.Context, cfg runtime.Config) {
 
 	// Register new checks
 	for _, c := range newChecks {
-		err = cc.RegisterCheck(ctx, c)
+		cc.RegisterCheck(ctx, c)
 		if err != nil {
 			log.ErrorContext(ctx, "Failed to register check", "check", c.Name(), "error", err)
 		}
```
Comment (lvlcn-t) [inline pkg/sparrow/controller.go:129]: You can remove the error check here

File: pkg/sparrow/controller_test.go [hunk 2]
```diff
@@ -236,69 +231,57 @@ func TestChecksController_Reconcile(t *testing.T) {
 
 func TestChecksController_RegisterCheck(t *testing.T) {
 	tests := []struct {
-		name    string
-		setup   func(t *testing.T) *ChecksController
-		check   checks.Check
-		wantErr bool
+		name  string
+		setup func(t *testing.T) *ChecksController
```
Comment (lvlcn-t) [inline pkg/sparrow/controller_test.go:235]: you can remove the `t *testing.T` argument here.

File: pkg/sparrow/controller_test.go [hunk 3]
```diff
@@ -236,69 +231,57 @@ func TestChecksController_Reconcile(t *testing.T) {
...
 }
 
 func TestChecksController_UnregisterCheck(t *testing.T) {
 	tests := []struct {
-		name    string
-		setup   func(t *testing.T) *ChecksController
-		check   checks.Check
-		wantErr bool
+		name  string
+		setup func(t *testing.T) *ChecksController
```
Comment (lvlcn-t) [inline pkg/sparrow/controller_test.go:269]: > you can remove the `t *testing.T` argument here.

Here aswell

File: pkg/sparrow/run.go [hunk 4]
```diff
@@ -172,7 +172,7 @@ func (s *Sparrow) shutdown(ctx context.Context) {
 		}
 		sErrs.errAPI = s.api.Shutdown(ctx)
 		s.loader.Shutdown(ctx)
-		sErrs.errCheckCont = s.controller.Shutdown(ctx)
```
Comment (lvlcn-t) [inline pkg/sparrow/run.go:175]: You can also remove the field from the error struct.

**Review by lvlcn-t — CHANGES_REQUESTED** [review]
Some minor things

File: pkg/sparrow/controller_test.go (hunk 3 above)
Comment (puffitos) [reply pkg/sparrow/controller_test.go]: I've removed the arguments from one of the tests and I've completely removed the function in the other test, as it was returning the same thing in both cases of the test suite.

File: pkg/sparrow/controller_test.go [hunk 5]
```diff
@@ -236,79 +231,60 @@ func TestChecksController_Reconcile(t *testing.T) {
...
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			cc := tt.setup(t)
+			cc := tt.setup()
 
-			err := cc.RegisterCheck(context.Background(), tt.check)
-			if (err != nil) != tt.wantErr {
-				t.Errorf("RegisterCheck() error = %v, wantErr %v", err, tt.wantErr)
-			}
+			cc.RegisterCheck(context.Background(), tt.check)
```
Comment (lvlcn-t) [inline pkg/sparrow/controller_test.go:261]: We should check for something in this test, otherwise we only test if it runs succussfully

Comment (puffitos) [reply pkg/sparrow/controller_test.go:261]: @lvlcn-t The scope of this PR is to remove the error from the controller. Let's not change the tests for now (which are fragile at best). We can open an issue regarding the tests.

Comment (lvlcn-t) [reply pkg/sparrow/controller_test.go:261]: Yeah, let's open an issue.

**Review by lvlcn-t — APPROVED** [review]

File: pkg/sparrow/controller_test.go (hunk 5 above)
Comment (puffitos) [reply pkg/sparrow/controller_test.go:261]: https://github.com/caas-team/sparrow/issues/131 done, will post PR soon to fix

**Pull request merged**